api/provenance/provenance.go

```diff
@@ -62,6 +62,13 @@ func GetProvenance() Provenance {
 			p.GitCommit = setting.Value
 		}
 	}
+
+	for _, dep := range info.Deps {
+		if dep.Path == "sigs.k8s.io/kustomize/kustomize/v5" {
+			p.Version = dep.Version
```

Comment:
If we are returning `p.Version` directly, we need to be aware of pseudo-versions too (https://github.com/golang/go/issues/50603). We can sanitize it further to figure out the closest tag as done here (https://github.com/operator-framework/helm-operator-plugins/blob/d00d0092fe5c2a7776a678b7dcee4085cbccfe01/internal/version/version.go#L85). If we are comfortable with printing it as is, then that also seems good. It would be nice to add a comment in case we want to revisit it later.